src/agent/debuglet.ts

```diff
@@ -106,6 +112,44 @@ const formatBreakpoints = function(
           .join('\n');
 };
 
+/**
+ * CachedPromise class is designed to support debug agent in Google Cloud
```

Comment:
IMO this description fits `Debuggee#isReady` better than `CachedPromise`. `CachedPromise` doesn't have anything to do with GCF on its own; it's a general-purpose construct that allows us to wait for a locally cached resource to be refreshed if it's deemed stale.

Reply:
Reorganized the comments.